### PR TITLE
Add redis as cache backend option

### PR DESCRIPTION
**Pull request by nipeone**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -103,6 +103,10 @@ commands:
           name: start memcached
           command: |
             docker run --rm -d -p 11211:11211 memcached -m 64
+      - run:
+          name: start redis
+          command: |
+            docker run --rm -d -p 6379:6379 redis -m 64
```
Comment (manthey) [inline .circleci/config.yml:109]: I think CI will pass if you remove the `-m 64` from the end of this line.

**Comment by manthey** [comment]
This is great.  With this, to pass tests locally one needs memcached and redis running.  I'm thinking that after this is merged, I'll add some environment checks so that the memcached and redis tests are only run if the appropriate service exists.

File: .circleci/config.yml (hunk 1 above)
Comment (nipeone) [reply .circleci/config.yml]: I'm very sorry it's taken so long to deal with this issue, I'll fix it and resubmit the PR

**Comment by manthey** [comment]
Thanks for working on this.  You should be able to fix the linting errors by doing `tox -e format`.

**Comment by nipeone** [comment]
i get this error like async call, but i'm sure i used sync function of redis. due to i'm not very familiar with CI.  maybe i need some help.
```
large_image/cache_util/rediscache.py:60: error: Argument 1 to "len" has incompatible type "Union[Awaitable[Any], Any]"; expected "Sized"  [arg-type]
large_image/cache_util/rediscache.py:65: error: Incompatible return value type (got "Union[Awaitable[Any], Any]", expected "bool")  [return-value]
large_image/cache_util/rediscache.py:79: error: Argument 1 to "loads" has incompatible type "Union[Awaitable[Any], Any]"; expected "Buffer"  [arg-type]
large_image/cache_util/rediscache.py:114: error: Incompatible return value type (got "Union[Awaitable[Any], Any]", expected "int")  [return-value]
large_image/cache_util/rediscache.py:143: error: Value of type "Union[Awaitable[Any], Any]" is not indexable  [index]
large_image/cache_util/rediscache.py:151: error: List or tuple expected as variadic arguments  [misc]
```

**Comment by manthey** [comment]
I think redis doesn't type response from the client aside from a very general type, so I'd use some judicious casts.  From your current checkin, I think the following diff would clean up the type and lint issues:
[diff.txt](https://github.com/girder/large_image/files/15420579/diff.txt)

**Review by manthey — APPROVED** [review]

**Comment by manthey** [comment]
@nipeone Thank you so much for adding this.  I'll make a follow on PR to make it so the redis tests are run conditionally to make it easier to test locally if redis isn't available.

**Pull request merged**